src/components/exchange/ExchangeField.js

```diff
@@ -85,47 +88,45 @@ const ExchangeField = (
 
   return (
     <Container {...props}>
-      <TouchableWithoutFeedback onPress={handleFocusField}>
+      <TouchableWithoutFeedback
+        onPress={onTapWhileDisabled || handleFocusField}
+      >
         <FieldRow disableCurrencySelection={disableCurrencySelection}>
           {symbol ? (
-            <CoinIcon address={address} symbol={symbol} type={type} />
+            <CoinIcon
+              address={address}
+              mainnet_address={mainnet_address}
+              symbol={symbol}
+              type={type}
+            />
           ) : (
             <CoinIconSkeleton />
           )}
-          {!editable && onTapWhileDisabled && (
-            <Text
-              color={amount > 0 ? colorForAsset : placeholderTextColor}
-              letterSpacing="roundedTightest"
-              size="h2"
-              weight="semibold"
-            >
-              {amount || placeholderText}
-            </Text>
-          )}
-          {editable && (
-            <Input
-              color={colorForAsset}
-              editable={editable}
-              onBlur={onBlur}
-              onChangeText={setValue}
-              onFocus={onFocus}
-              onTapWhileDisabled={onTapWhileDisabled}
-              placeholder={placeholderText}
-              placeholderTextColor={placeholderTextColor}
-              ref={ref}
-              testID={amount ? `${testID}-${amount}` : testID}
-              useCustomAndroidMask={useCustomAndroidMask}
-              value={amount}
-            />
-          )}
+
+          <Input
+            color={colorForAsset}
+            editable={editable}
+            onBlur={onBlur}
+            onChangeText={setValue}
+            onFocus={onFocus}
+            placeholder={placeholderText}
+            placeholderTextColor={placeholderTextColor}
+            {...(onTapWhileDisabled && { pointerEvents: 'none' })}
+            ref={ref}
+            testID={testID}
+            useCustomAndroidMask={useCustomAndroidMask}
+            value={amount}
+          />
         </FieldRow>
       </TouchableWithoutFeedback>
       {!disableCurrencySelection && (
         <TokenSelectionButton
           address={address}
+          mainnet_address={mainnet_address}
```

Comment:
you're going to hell for writing a prop like this

Reply:
in my defense u started it